src/nvim/event/libuv_process.c

```diff
@@ -101,6 +101,16 @@ static void close_cb(uv_handle_t *handle)
 static void exit_cb(uv_process_t *handle, int64_t status, int term_signal)
 {
   Process *proc = handle->data;
-  proc->status = (int)status;
+#if defined(WIN32)
+  // Use stored/expected signal.
+  // XXX: this might actually be off, e.g. when the process catches SIGTERM
+  // itself, and exits normally then.
+  term_signal = proc->exit_signal;
```

Comment:
The comment is confusing to me. Isn't it obvious? Maybe the purpose of `exit_signal` should be better explained (in a comment at `struct process`).

Reply:
Removed the (additional "XXX") comment.
The struct has a comment already - can it be clearer.  Maybe also here?
The point is that for Windows we're using the signal that we send ourselves, which might be not what was really used.

I've thought that we could have a test in this regard that would send a signal, but then trap it, and exit instead - this should not have term_signal on Unix then, but the used exit signal on Windows.